specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/SessionPools.json

```diff
@@ -0,0 +1,702 @@
+          "$ref": "#/definitions/SessionContainerResources",
+          "description": "Container resource requirements."
+        }
+      }
+    },
+    "SessionIngress": {
+      "description": "Session Pool Ingress configuration.",
+      "type": "object",
+      "properties": {
+        "targetPort": {
+          "format": "int32",
+          "description": "Target Port in containers for traffic from ingress",
+          "type": "integer"
+        }
+      }
+    },
+    "CustomContainerTemplate": {
+      "description": "Custom container configuration.",
+      "type": "object",
+      "properties": {
+        "registryCredentials": {
+          "description": "Private container registry credentials for containers used by the Sessions of the Session Pool.",
+          "$ref": "#/definitions/SessionRegistryCredentials"
+        },
+        "containers": {
+          "description": "List of container definitions for the Sessions of the Session Pool.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SessionContainer"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "ingress": {
+          "description": "SessionPool Ingress configuration.",
+          "$ref": "#/definitions/SessionIngress"
+        }
+      }
+    },
+    "SessionNetworkConfiguration": {
+      "description": "Session network configuration.",
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "Network status for the Sessions.",
+          "type": "string",
+          "enum": [
+            "egressEnabled",
+            "egressDisabled"
+          ],
+          "x-ms-enum": {
+            "name": "SessionNetworkStatus",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "SessionPool": {
+      "description": "Container App Session Pool.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Container App Session Pool resource specific properties",
+          "type": "object",
+          "properties": {
+            "environmentId": {
+              "description": "Resource ID of the Session Pool's environment.",
+              "type": "string",
+              "format": "arm-id",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.App/managedEnvironments"
+                  }
+                ]
+              },
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            },
+            "containerType": {
+              "description": "The container type of the Sessions.",
+              "enum": [
+                "python-lts",
+                "node-lts"
```

Comment:
existing enums 

"JupyterPython",
 "FunctionsPython",
"CustomContainer"

are not required anymore?

Reply:
Yes, we are changing these to the current one

Reply:
why enum values are not following `Pascal Casing` pattern ? If there are well known values then it should be okay.